org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/participants/DTDErrorCode.java

```diff
@@ -154,7 +156,24 @@ public static Range toLSPRange(XMLLocator location, DTDErrorCode code, Object[]
 		case PEReferenceWithinMarkup: {
 			return XMLPositionUtility.getLastValidDTDDeclParameter(offset, document, true);
 		}
-		
+		case EntityNotDeclared: {
+			TextDocument textDocument = document.getTextDocument();
+			try {
+				Position position = textDocument.positionAt(offset);
+				int line = position.getLine();
+				String text = textDocument.lineText(line);
+				String name = (String) arguments[0];
+				String subString = "&" + name;
+				int start = text.indexOf(subString) + 1;
+				int end = start + name.length();
+				Position startPosition = new Position(line, start);
+				Position endPosition = new Position(line, end);
+				return new Range(startPosition, endPosition);

```

Comment:
note for later: we should have a `public static Range newRange(line, start, end)` utility method somewhere

Reply:
should this change take place in a different PR?

Reply:
definitely yes

Reply:
XMLPositionUtility is the best place to add this method